From 4c9f8814eb9b4d4b2592d05fefab146496b5d8a2 Mon Sep 17 00:00:00 2001
From: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
Date: Thu, 25 May 2023 10:02:31 +0600
Subject: [PATCH 29/85] arm64: dts: renesas: r8a779f0: add intswcd node

Since intswcd is a control domain device, the node is added to
r8a779f0-ctrl-domain.dtsi file.

Signed-off-by: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
---
 arch/arm64/boot/dts/renesas/r8a779f0-ctrl-domain.dtsi | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a779f0-ctrl-domain.dtsi b/arch/arm64/boot/dts/renesas/r8a779f0-ctrl-domain.dtsi
index 9c952cdec9e1..9bbe3375470a 100644
--- a/arch/arm64/boot/dts/renesas/r8a779f0-ctrl-domain.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a779f0-ctrl-domain.dtsi
@@ -1,3 +1,5 @@
+#include <dt-bindings/interrupt-controller/renesas-intswcd-r8a779f0.h>
+
 &soc {
 	gpio4: gpio@dfd90180 {
 		compatible = "renesas,gpio-r8a779f0";
@@ -42,6 +44,15 @@ gpio7: gpio@dfd91980 {
 		power-domains = <&sysc R8A779F0_PD_ALWAYS_ON>;
 		resets = <&cpg 915>;
 	};
+
+	intswcd: intswcd@d8f11000 {
+		compatible = "renesas,intswcd-r8a779f0";
+		reg = <0 0xd8f11000 0 0x400>;
+		first-level-interrupt = <&gic GIC_SPI 522 IRQ_TYPE_LEVEL_HIGH>;
+		first-edge-interrupt = <&gic GIC_SPI 573 IRQ_TYPE_LEVEL_HIGH>;
+		interrupt-controller;
+		#interrupt-cells = <1>;
+	};
 };
 
 /* To enable control comain in cpg driver, need to define second reg range.
-- 
2.39.2

